pkg/manifests/configfile.go

```diff
@@ -23,15 +24,79 @@ const (
 	CommandTimeout = time.Minute
 )
 
+// This is easier to read as YAML, trust me.
+const configSchemaYAML = `
+"$schema": http://json-schema.org/draft-07/schema#
+definitions:
+  command:
+    type: object
+    required: ['command']
+  version: { const: 1 }
+type: object
+oneOf:
+- required: ['version', 'commandUpdated']
+  properties:
+    version: { '$ref': '#/definitions/version' }
+    commandUpdated:
+      required: ['generators']
+      properties:
+        generators:
+          type: array
+          items: { '$ref': '#/definitions/command' }
+        updaters:
+          type: array
+          items:
+            type: object
+            properties:
+              containerImage: { '$ref': '#/definitions/command' }
+              policy: { '$ref': '#/definitions/command' }
+      additionalProperties: false
+- required: ['version', 'patchUpdated']
+  properties:
+    version: { '$ref': '#/definitions/version' }
+    patchUpdated:
+      required: ['generators', 'patchFile']
+      properties:
+        patchFile: { type: string }
+        generators:
+          type: array
+          items: { '$ref': '#/definitions/command' }
+      additionalProperties: false
+- required: ['version', 'files']
+  properties:
+    version: { '$ref': '#/definitions/version' }
+    files:
+      additionalProperties: false
+  additionalProperties: false
+`
+
+var ConfigSchema *jsonschema.Schema
+
+func init() {
```

Comment:
Instead of using `init()` could you rename this to `mustCompileSchema(string)` (or the likes) and do:

```
var ConfigSchema *jsonschema.Schema = mustCompileSchema(configSchemaYAML)
```

Reply:
Sure. I presume it would compile to the same thing, is there a reason to prefer this formulation?

Reply:
`init()` functions play a special role in go, to avoid confusion I would rename it.

Reply:
@stefanprodan Yes, I know, I am using it for the special role.

Reply:
> is there a reason to prefer this formulation?

It is more explicit and readable

Reply:
okdokes